### PR TITLE
feat: add support for multiple OTEL exporters

### PR DESCRIPTION
**Pull request by fiam**
Support both OTELHTTP and OTELGRPC too

The resulting configuration looks like:

```yaml
telemetry:
    tracing:
        enabled: true
        exporters:
            - endpoint: http://localhost:4317
              exporter: otlpgrpc
              headers: {Authorization: Bearer ${GRAPH_API_TOKEN}}
    metrics:
        enabled: true
        opentelemetry:
            enabled: true
            exporters:
                - endpoint: http://localhost:4317
                  exporter: otlpgrpc
                  headers: {Authorization: Bearer ${GRAPH_API_TOKEN}}
```

Backwards compatibility is preserved, keeping the old values at the root of the telemetry/metrics fields to configure a single endpoint. Additionally, a boolean `use_default_exporters` parameter has been added, to allow disabling the default exporters.

## Motivation and Context

I was asked nicely

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: router/internal/config/config.go [hunk 1]
```diff
@@ -49,22 +63,43 @@ type Prometheus struct {
...
 }
 
-type OpenTelemetry struct {
-	ServiceName string            `yaml:"service_name" default:"cosmo-router" envconfig:"TELEMETRY_SERVICE_NAME" validate:"required"`
-	Endpoint    string            `yaml:"endpoint" validate:"required" default:"https://cosmo-otel.wundergraph.com" envconfig:"TELEMETRY_ENDPOINT" validate:"http_url"`
-	Headers     map[string]string `yaml:"headers" envconfig:"TELEMETRY_HEADERS"`
-
-	Tracing Tracing `yaml:"tracing"`
-	Metrics Metrics `yaml:"metrics"`
+// telemetry:
```
Comment (StarpTech) [inline router/internal/config/config.go:83]: Can we delete it?

File: router/internal/metric/agent.go [hunk 2]
```diff
@@ -31,33 +35,65 @@ func createPromExporter() (*prometheus.Exporter, error) {
...
+		}
+		if len(exp.HTTPPath) > 0 {
+			opts = append(opts, otlpmetrichttp.WithURLPath(exp.HTTPPath))
+		}
+
+		exporter, err = otlpmetrichttp.New(
+			context.Background(),
+			opts...,
+		)
+	case otelconfig.ExporterOLTPGRPC:
```
Comment (StarpTech) [inline router/internal/metric/agent.go:67]: Did you test GRPC integration?

File: router/internal/trace/config.go [hunk 3]
```diff
@@ -1,41 +1,52 @@
...
 	// For example
 	// /v1/traces
-	OtlpHttpPath string
+	HTTPPath string
+}
+
+// Config represents the configuration for the agent.
+type Config struct {
+	Enabled bool
+	// Name represents the service name for tracing. The default value is wundergraph-cosmo-router.
```
Comment (StarpTech) [inline router/internal/trace/config.go:31]: It's just `cosmo-router`

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router/internal/metric/agent.go (hunk 2 above)
Comment (fiam) [reply router/internal/metric/agent.go:67]: I did, this does send the metrics over gRPC:

```yaml
telemetry:
    use_default_exporters: false
    tracing:
        enabled: true
        exporters:
            - endpoint: http://localhost:4317
              exporter: otlpgrpc
```

File: router/internal/trace/config.go (hunk 3 above)
Comment (fiam) [reply router/internal/trace/config.go]: I just moved that around 😅 - but fixed, which is what matters!

File: router/internal/config/config.go (hunk 1 above)
Comment (fiam) [reply router/internal/config/config.go]: Sure, I `git add` ed it by mistake, sorry about that!

**Comment by fiam** [comment]
Pending docs update

**Comment by fiam** [comment]
Docs added

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**